### PR TITLE
#2657 add insurance policy lookup record creation logic

### LINKED ISSUE #2657: Add patient API insurance policy creation logic
## Is your feature request related to a problem? Please describe.

When a patient is added to the local database using the lookup API, related insurance policy records also need to be created.

## Describe the solution you'd like

Add lookup API logic for creating insurance policy records.

## Implementation

Add insurance policy record creation logic to existing `lookupApi` helpers.

## Describe alternatives you've considered

N/A.

## Additional context

See epic #2446.

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #2657.

## Change summary

Adds insurance policies for patient records added via lookup API.

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] When a prescription is made for a patient retrieved using the API, all associated insurance policies appear in the dropdown list.

### Related areas to think about

Should finish up the creation logic, but a lot of duplication occurring. Next steps are to extract out some shared logic from `createRecord` and `incomingSyncUtils`.

### CODE REVIEW

File: src/utilities/network/lookupApi.js [hunk 1]
```diff
@@ -73,6 +102,7 @@ export const createPatientRecord = response => {
     isVisible: true,
   };
   UIDatabase.write(() => UIDatabase.update('Name', patient));
+  policies.forEach(policy => createInsurancePolicyRecord(policy));
```
Comment (mark-prins) [inline src/utilities/network/lookupApi.js:105]: Just for fun and to remove a tiny bit of code:

```suggestion
  policies.forEach(createInsurancePolicyRecord);
```

File: src/utilities/network/lookupApi.js [hunk 2]
```diff
@@ -156,11 +186,32 @@ const getPrescriberRequestUrl = params => {
...
-  const patientData = responseJson.map(
+const processInsuranceResponse = response => response.map(({
+    ID: id,
+    insuranceProviderID: insuranceProviderId,
+    nameID: nameId,
+    policyNumberFamily,
+    policyNumberPerson,
+    discountRate,
+    expiryDate,
+    isActive: isActive,
```
Comment (mark-prins) [inline src/utilities/network/lookupApi.js:197]: unnecessary rename; should just be `isActive`

File: src/utilities/network/lookupApi.js [hunk 3]
```diff
@@ -22,6 +22,34 @@ const SEPARATORS = {
   POLICY_NUMBER: '-',
 };
 
+export const createInsurancePolicyRecord = response => {
```
Comment (joshxg) [inline src/utilities/network/lookupApi.js:25]: same comment as before - adding a case to `createRecord` should be what we're doing here!

File: src/utilities/network/lookupApi.js [hunk 4]
```diff
@@ -22,6 +22,34 @@ const SEPARATORS = {
...
+    type,
+    discountRate: parseNumber(discountRate),
+    expiryDate: parseDate(expiryDate),
+    enteredBy: UIDatabase.getOrCreate('User', enteredById),
+    patient: UIDatabase.getOrCreate('Name', nameId),
+    insuranceProvider: UIDatabase.getOrCreate('InsuranceProvider', insuranceProviderId),
+    isActive: parseBoolean(isActive)
+  };
+  UIDatabase.write(() => UIDatabase.update('InsurancePolicy', policy));
+};
```
Comment (joshxg) [inline src/utilities/network/lookupApi.js:51]: also if we're integrating insurance policies from other stores we need a mechanism to ensure they're read only. An easy solution would be to add a getter to `InsurancePolicy`?

```
get canEdit(){
    return this.patient.thisStoresPatient
}
```

and then obviously the code on the policy edit to ensure it's read only?

Is there any other patient data being integrated into mobile?

Comment (joshxg) [reply src/utilities/network/lookupApi.js:51]: and possibly to be extra safe adding a check in outgoing sync to ensure policies which aren't from this stores patients are never synced out

Comment (wlthomson) [reply src/utilities/network/lookupApi.js:51]: > also if we're integrating insurance policies from other stores we need a mechanism to ensure they're read only

#2543 :).
 
> and possibly to be extra safe adding a check in outgoing sync to ensure policies which aren't from this stores patients are never synced out

#2544 :).

File: src/utilities/network/lookupApi.js (hunk 3 above)
Comment (wlthomson) [reply src/utilities/network/lookupApi.js:25]: #2664 :).

File: src/utilities/network/lookupApi.js (hunk 1 above)
Comment (wlthomson) [reply src/utilities/network/lookupApi.js:105]: Fixed in #2671 (fcdd2832).

File: src/utilities/network/lookupApi.js (hunk 2 above)
Comment (wlthomson) [reply src/utilities/network/lookupApi.js:197]: Oops, fixed in #2671.

**Pull request merged**